### PR TITLE
Enable re-reblocking with no PLs

### PR DESCRIPTION
**Pull request by ldgauthier**
For example. some gnomAD 3.1 samples were reblocked in production using the higher resolution schema, but needed to be reblocked again to be consistent with the gnomAD 3.0 schema, but PLs were removed for compression's sake.

### CODE REVIEW

**Comment by droazen** [comment]
@ldgauthier Could you nominate a reviewer for this one?

**Comment by ldgauthier** [comment]
Maybe @vruano is a good person to review so we spread some knowledge of the joint calling pipeline.

File: src/main/java/org/broadinstitute/hellbender/utils/variant/writers/HomRefBlock.java [hunk 1]
```diff
@@ -124,6 +128,15 @@ public void add(final int pos, final int newEnd, final Genotype genotype) {
                 }
             }
         }
+        if (minPPs != null) {
+            minGQ = GATKVariantContextUtils.calculateGQFromPLs(minPPs);
+        } else if (minPLs != null) {
+            minGQ = GATKVariantContextUtils.calculateGQFromPLs(minPLs);
+        } else if (minGQ == -1) {
+            minGQ = genotype.getGQ();
```
Comment (vruano) [inline src/main/java/org/broadinstitute/hellbender/utils/variant/writers/HomRefBlock.java:136]: I would compact the two last elses as minGQ = minGQ == -1 ? genotype.getGQ() : Math.min(...);

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCFIntegrationTest.java [hunk 2]
```diff
@@ -185,4 +185,19 @@ public void testMQHeadersAreUpdated() throws Exception {
...
+        final File output = createTempFile("rereblockedgvcf", ".vcf");
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.add("V", input)
+                .addOutput(output);
+        runCommandLine(args);
+
+        Pair<VCFHeader, List<VariantContext>> inputVCs = VariantContextTestUtils.readEntireVCFIntoMemory(input.getAbsolutePath());
+        Pair<VCFHeader, List<VariantContext>> outputVCs = VariantContextTestUtils.readEntireVCFIntoMemory(output.getAbsolutePath());
+
+        Assert.assertTrue(inputVCs.getRight().size() > outputVCs.getRight().size());
```
Comment (vruano) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCFIntegrationTest.java:201]: This test seems a bit weak.... can you do a more through test e.g. the unaffected blocks should be in the same order and unchanged... etc.

File: src/test/java/org/broadinstitute/hellbender/utils/variant/writers/HomRefBlockUnitTest.java [hunk 3]
```diff
@@ -110,6 +110,22 @@ public void testCantAddDifferentNumbersOfPls(){
...
     }
 
+    @Test
+    public void testNoPLs() {
+        //add VC with no PLs to block with PLs
+        final VariantContext vc = getVariantContext();
+        final GVCFBlock band = getHomRefBlock(getVariantContext());
+        band.add(vc.getStart(), getValidGenotypeBuilder().make() );
+        band.add(vc.getStart() + 1, getValidGenotypeBuilder().noPL().make());
+        Assert.assertTrue(band.getSize() == 2);
```
Comment (vruano) [inline src/test/java/org/broadinstitute/hellbender/utils/variant/writers/HomRefBlockUnitTest.java:120]: Looks a bit weak of a test. Anything else that can be tested

**Review by vruano — APPROVED** [review]
Is it possible to make this work with non-diploids? In a separate issue?

**Pull request merged**